### PR TITLE
Signal number map

### PR DESCRIPTION
**Pull request by treitter**
Support signal number map files

### CODE REVIEW

File: vsm [hunk 1]
```diff
@@ -65,12 +66,15 @@ config_tree = None
...
+signal_to_num = {}
 
 
 def _format_signal_msg(signal, value, indicator):
-    # NOTE: the "[SIGNUM]" field is a temporary placeholder for the signal ID
-    return '{} {},{},[SIGNUM],{}'.format(indicator, get_runtime(),
-                                         signal, repr(value))
+    signum = "[SIGNUM]"
+    if signal in signal_to_num:
+        signum = str(signal_to_num[signal])
```
Comment (kov) [inline vsm]: I guess you don't need the str() here as .format() can take ints?

File: vsm [hunk 2]
```diff
@@ -65,12 +66,15 @@ config_tree = None
...
+    return '{} {},{},{},{}'.format(indicator, get_runtime(),
+                                         signal, signum, repr(value))
```
Comment (kov) [inline vsm]: indentation is a bit odd in some places

File: vsm [hunk 3]
```diff
@@ -193,6 +200,11 @@ class State(object):
 
         return ast_node
 
+    def _exit_signal_num_missing(self, signal):
+        print("signal '{}' not in signal number mapping file".format(signal),
+                file=sys.stderr)
```
Comment (kov) [inline vsm]: indentation is off compared to what's on the other lines

File: vsm [hunk 4]
```diff
@@ -910,8 +926,14 @@ if __name__ == "__main__":
     parser.set_defaults(log_condition_checks=True)
     parser.add_argument('--log-format', choices=['catapult'],
                         help='Write log file in specified format')
+    parser.add_argument('--signal-number-file', type=str,
+            help='A .vsi file which maps all signal names to numbers',
```
Comment (kov) [inline vsm]: Indentation is different from the lines around it

File: tests.py [hunk 5]
```diff
@@ -147,126 +169,126 @@ def run_vsm(self, name, input_data, expected_output,
...
 State = {
 car.stop = True
 damage = True
 moving = false
 }
 condition: (moving != True and damage == True) => True
-damage,[SIGNUM],'true'
-car.stop,[SIGNUM],'True'
+damage,5,'true'
+car.stop,4,'True'
```
Comment (kov) [inline tests.py]: It's a bit weird that some are true and some are True, I guess I know why that is, but maybe we should add a change to normalise that?

**Review by kov — APPROVED** [review]
Looks good, just a few nits. so I'm setting it to approved already.

File: vsm (hunk 1 above)
Comment (treitter) [reply vsm]: Fixed. I was over-compensating for the fact that Python fails to do this automatically in most other cases.

File: vsm (hunk 4 above)
Comment (treitter) [reply vsm]: Fixed this and the other indentation inconsistencies. I was just using vim's default indentation which I misunderstood to be the same as the Python style guide.

File: tests.py (hunk 5 above)
Comment (treitter) [reply tests.py]: I've done so in this latest version

**Pull request merged**